### DISCUSSION TITLE
BUG? Audacious has problems loading/playing valid audio files, added via symlinks

### DISCUSSION THREAD

**Discussion by jp9753** [audacious-media-player/audacious#95] [General]
I couldn't find a report a bug section, can somebody move this to the 'report a bug section' if it exists, thanks

Audacious 3.10.01 (using Winamp Classic interface)

Linux Mint 20.3 (XFCe desktop)
All Browsing, highlighting, dragging & dropping performed using Thunar (4.16.8)
Thunar is the XFCe File Manager

I have some (music) folders containing symlinks pointing at 'music tracks' stored in other folders.
This allows easy playback of albums in a preferred order. But Audacious doesn't handle symlinks correctly. 

TEST_1 (works as expected)
start Audacious
make sure playlist is empty
browse into folderA (containing audio files)
highlight audio files
drag and drop files onto the Audacious Playlist
Audacious accepts the files and adds the files (from folderA) to the Playlist
Automatic playback starts

TEST_2 (almost works as expected)
start Audacious
make sure playlist is empty
browse into folderB (only containing symlinks to each audio file in folderA)
highlight the symlinks (folderB)
drag and drop symlinks onto the Audacious Playlist
Audacious accepts the symlinks (from folderB) and adds the files (from folderA) to the Playlist
Automatic playback DOES NOT start (pressing play will start playback)

(a minor bug maybe, but Audacious' actions (in TEST_2) should match its actions (in TEST_1)
If playback starts in TEST_1, playback also should start in TEST_2

TEST_3 (works as expected)
start Audacious
make sure playlist is empty
browse to parent folder of folderA
highlight folderA
drag and drop folderA onto Audacious Playlist
Audacious accepts the folder and adds the files (contents of folderA) to the Playlist
Automatic playback starts

TEST_4 (FAILURE)
start Audacious
make sure playlist is empty
browse to parent folder of folderB
highlight folderB (which contains the symlinks)
drag and drop folderB into Audacious Playlist
Audacious accepts the folder, but DOES NOT add the files (from folderA) to the Playlist
Automatic playback DOES NOT start (empty playlist)

A MAJOR BUG

I repeated (TESTS_1~4) in VLC and all worked correctly (ie loaded the files and started playing)
So it's not a file manager issue.

NOTE: please don't suggest I can achieve the same effect with playlists, etc. that's not the issue.
It would be nice if Audacious just handled symlinks correctly.

**Comment by jlindgren90** [comment]
You are using a very old version of Audacious (from 2018). Try the current version.

**Comment by jp9753** [comment]
that's the version in the Linux Mint 20.3 repository.
Mint is debian based (forked from Ubuntu), what's the best scheme for
pointing to a new version?

so I can sudo apt install audacious

kind regards
simon

**Comment by jlindgren90** [comment]
Follow the instructions for Debian derivatives on our website, audacious-media-player.org

**Comment by jp9753** [comment]
followed the instructions       sudo apt-get install audacious

and got    'audacious is already the newest version (3.10.1-1build1)'

with reference to the note on Panda Jim's PPA  (referenced on your page
(audacious-media-player.org <http://audacious-media-player.org>) under the
apt-get install audacious instruction)
------
Some weeks ago, I emailed Panda Jim with several questions, but I never
received an answer

Q1 for Panda Jim was) if there were any known issues installing the 'newer'
audacious versions (from a PPA intended for Ubuntu)?

as I understand it, Audacious, needs to have been compiled for Ubuntu
20.04.5 LTS (Focal Fossa) (Mint 20.3's fork)

I found these entries in the PPA list (I presume this is meant to be 20.04
and isn't the year 2004 )
audacious                                       4.3.1-0build1~ubuntu2004
audacious-plugins                          4.3.1-0build1~ubuntu2004

Q2 for Panda Jim was) is this 20.04.5 LTS (Focal Fossa) or is it the year
2004?

Q3 for Panda Jim was) how do I limit the impact of using a Ubuntu PPA, so I
only install Audacious

kind regards
simon

**Comment by radioactiveman** [comment]
You should find all information regarding the installation in Jims [blog post](https://ubuntuhandbook.org/index.php/2023/03/audacious-music-player-4-3-ppa/). I would recommend to use the PPA approach.

To answer your questions:
Q1: Jim mentions Linux Mint in his blog, so I assume there shouldn't be a problem. Just try it out. Otherwise remove the PPA again and rollback to the original version.
Q2: It's the version built for Ubuntu 20.04 (20.04 stands for April 2020, not 2004). Your package manager will take care of it, you can only install one specific version.
Q3: The number of packages in this PPA is not that high, especially if you consider all the Ubuntu versions. Make sure to update your system before adding the PPA. Then you can see which packages are actually installed/upgraded from the PPA.

**Comment by jp9753** [comment]
thank you Thomas & John,
I get a little nervous updating 'Operating Systems'. Once, I could identify
(and understood the usage of) every single file in PC-DOS 3.30, then came
Windows 3.0, when the mere act of installing software could/did have a long
term impact on the increasingly (bloatware) OS's. Many years of personnel
usage (plus being the office 'IT expert' in residence), taught me you
really need a VERY GOOD reason to make any changes, along with a little
research beforehand, (hopefully,) stops really stupid decisions.
The devastation caused, by the 'user', installing 'the wrong versions' of
(legit) software, has in my experience, always been far worse than any
Virus or Malware!

kind regards
simon

**Comment by mvtiaine** [comment]
I guess the OP issue was about symlinked files, but relatedly Audacious currently explicitly does not support symlinked folders:
https://github.com/audacious-media-player/audacious/blob/master/src/libaudcore/adder.cc#L354

It would be nice to support those as well, perhaps enabled via conf option.
My use case is for example with local Modland mirror (https://ftp.modland.com/pub/modules/), I'd like to be able to define a subset of formats with symlinks like:
modules.amiga/AHX -> ../modules/AHX
modules.amiga/AM Composer -> ../modules/AM Composers
...
